src/coreclr/jit/morphblock.cpp

```diff
@@ -470,6 +470,9 @@ GenTree* MorphInitBlockHelper::MorphCommaBlock(Compiler* comp, GenTreeOp* firstC
     {
         GenTree* comma = commas.Pop();
         comma->gtType  = TYP_BYREF;
+
+        // The "IND(COMMA)" => "COMMA(IND)" transform may have set NO_CSEs on these COMMAs, clear them.
```

Comment:
Why would the transformation set NO_CSEs on the COMMAs?

Reply:
It [copies](https://github.com/dotnet/runtime/blob/9656b4760f01d3933e799ddf4c4d94649c6eae11/src/coreclr/jit/morph.cpp#L12099-L12100) them from the `IND`. The exact reason has been lost to history, but the end result is that:

a) For `IND`s that are location nodes (LHS of `ASG` / operand of `ADDR`) the invariant that they must have `NO_CSE` set is maintained.
b) Deficiencies in our CSE algorithm are worked around - one will get a good number of regressions if one deletes that copying.

It's notable that the `NO_CSE` flag has four purposes in the compiler:
1) Working around issues with constant and/or copy propagation.
2) Working around issues with inability to determine whether a given node is a "location" or value.
3) Working around issues with the CSE algorithm (including those that are by design).
4) Preserving previous behavior w.r.t. some block nodes (that used to be implicit).

So it can get set in a good number of situations; not always for great reasons.